tests/Sdk/Unit/Infrastructure/Builder/TaskSet/TaskSetTaskRelationsBuilderTest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Sdk\Unit\Infrastructure\Builder\TaskSet;
+
+use Codeception\Test\Unit;
+use InvalidArgumentException;
+use SprykerSdk\Sdk\Infrastructure\Builder\TaskSet\TaskSetTaskRelationsBuilder;
+use SprykerSdk\SdkContracts\Entity\TaskInterface;
+use SprykerSdk\SdkContracts\Entity\TaskSetInterface;
+
+/**
+ * Auto-generated group annotations
+ *
+ * @group Unit
+ * @group Infrastructure
+ * @group Builder
+ * @group TaskSet
+ * @group TaskSetTaskRelationsBuilderTest
+ * Add your own group annotations below this line
+ */
+class TaskSetTaskRelationsBuilderTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testBuildFromTaskSetShouldThrowExceptionWhenTaskSetNotInExistingTasks(): void
+    {
+        // Arrange
+        $taskSetMock = $this->createTaskSetMock('task-set:id', []);
+        $taskSetTaskRelationsBuilder = new TaskSetTaskRelationsBuilder();
+
+        $this->expectException(InvalidArgumentException::class);
+
+        // Act
+        $taskSetTaskRelationsBuilder->buildFromTaskSet($taskSetMock, []);
+    }
+
+    /**
+     * @return void
+     */
+    public function testBuildFromTaskSetShouldThrowExceptionWhenSubTaskIdNotInExistingTasks(): void
+    {
+        // Arrange
+        $taskSetMock = $this->createTaskSetMock('task-set:id', ['task:id']);
+        $taskSetTaskRelationsBuilder = new TaskSetTaskRelationsBuilder();
+        $existingTasks = ['task-set:id' => $taskSetMock];
+
+        $this->expectException(InvalidArgumentException::class);
```

Comment:
// Assert

The same for the method above and the rest of new tests with the exception checks.